src/Silex/ContentNegotiationProvider.php

```diff
@@ -61,31 +61,31 @@ public function register(Container $app)
         $app['negotiate.accept'] = function () {
             return function (string ...$types) {
                 return function (Request $request, Container $app) use ($types) {
-                    return $app['content_negotiator.accept']->negotiate($request, $types);
+                    $app['content_negotiator.accept']->negotiate($request, $types);
```

Comment:
Maybe `ContentNegotiator` should actually return the match and not set it on a request attribute? That could be done here instead.

Reply:
The current solution seems to minimize the code duplicated between different controllers. This part is not replicated so a change here should be neutral in that respect. If it improved readability here (multiple anonymous functions nested inside each other) it's a net positive.